pkg/image/api/v1/types.go

```diff
@@ -144,3 +182,44 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+type ImageStreamImport struct {
+	unversioned.TypeMeta `json:",inline"`
+	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
+
+	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
+	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
+}
+
+type RepositoryImport struct {
+	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
+	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
+
+	ImportPolicy    TagImportPolicy `json:"importPolicy,omitempty" description:"policy controlling how the image is imported"`
+	IncludeManifest bool            `json:"includeManifest,omitempty" description:"if true, return the manifest for each image in the response"`
+
+	RepositoryImportStatus `json:",inline" description:"status of the attempt to import images"`
+}
+
+type RepositoryImportStatus struct {
+	Status unversioned.Status `json:"status,omitempty" description:""`
+	// Images is a list of images successfully retrieved by the import of the repository.
+	Images []Image `json:"images,omitempty" description:"a list of the images retrieved by the import of the repository"`
+	// AdditionalTags is a list of tags that were not imported by this request
+	AdditionalTags []string `json:"additionalTags,omitempty" description:"a list of additional tags on the repository that were not retrieved"`
+}
+
+type ImageImport struct {
+	From kapi.ObjectReference       `json:"from" description:"the source of an image to import; only kind DockerImage is allowed"`
+	To   *kapi.LocalObjectReference `json:"to,omitempty" description:"a tag in the current image stream to assign the imported image to"`
```

Comment:
Matching tags as what?

On Mon, Jan 11, 2016 at 1:37 PM, David Eads <notifications@github.com>
wrote:

> In pkg/image/api/v1/types.go
> <https://github.com/openshift/origin/pull/6288#discussion_r49359013>:
>
> > +	IncludeManifest bool            `json:"includeManifest,omitempty" description:"if true, return the manifest for each image in the response"`
> > +
> > +	RepositoryImportStatus `json:",inline" description:"status of the attempt to import images"`
> > +}
> > +
> > +type RepositoryImportStatus struct {
> > +	Status unversioned.Status `json:"status,omitempty" description:""`
> > +	// Images is a list of images successfully retrieved by the import of the repository.
> > +	Images []Image `json:"images,omitempty" description:"a list of the images retrieved by the import of the repository"`
> > +	// AdditionalTags is a list of tags that were not imported by this request
> > +	AdditionalTags []string `json:"additionalTags,omitempty" description:"a list of additional tags on the repository that were not retrieved"`
> > +}
> > +
> > +type ImageImport struct {
> > +	From kapi.ObjectReference       `json:"from" description:"the source of an image to import; only kind DockerImage is allowed"`
> > +	To   *kapi.LocalObjectReference `json:"to,omitempty" description:"a tag in the current image stream to assign the imported image to"`
>
> Same as to == false
>
> Add to docs so that users will know. That's less obvious that it seems. I
> had expected the default behavior to be matching tags.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/6288/files#r49359013>.
>